### ISSUE TITLE
<blockquote> not allowed inside of <dd>

### ISSUE THREAD

**Issue by ietf-svn-bot** [ietf-tools/xml2rfc#570]
`type_defect`   |    by arusso@amsl.com

___

Example of where <blockquote> inside <dd> seems useful in the following text. 

(The workaround is to just use quotation marks, as in the original - which is readable; it's just not how the same quote would be handled if it were in a <t>.)

Original (draft-ietf-hip-native-nat-traversal-33):
```
   Locator:
      As defined in [RFC8046]: "A name that controls how the packet is
      routed through the network and demultiplexed by the end-host.  It
      may include a concatenation of traditional network addresses such
      as an IPv6 address and end-to-end identifiers such as an ESP
      Security Parameter Index (SPI).  It may also include transport
      port numbers or IPv6 Flow Labels as demultiplexing context, or it
      may simply be a network address."
```

Attempted to use <blockquote> - envisioning output along the lines of:
```
   Locator:
      As defined in [RFC8046]: 
      |  A name that controls how the packet is routed through the
      |  network and demultiplexed by the end host.  It may include a
      |  concatenation of traditional network addresses such as an IPv6
      |  address and end-to-end identifiers such as an ESP SPI.  It may
      |  also include transport port numbers or IPv6 Flow Labels as
      |  demultiplexing context, or it may simply be a network address.
```

File:
https://www.rfc-editor.org/v3test/test_blockquote.xml

yields
(234): Error: Element dd has extra content: blockquote, at /rfc/middle/section9ec338d81e4f22a396878fc78da1259c713489a2/dl/dd581394b850b8186c0bd36f7dddcae5d3ba5371e4/blockquote

___
_Issue migrated from trac:570 at 2022-02-08 07:11:06 +0000_

**Comment by alicerusso** [comment]
Another example:
Section 5.1 of draft-ietf-tsvwg-l4s-arch-20 contains a block quote from RFC 3168 within a ``<dd>`` (https://www.ietf.org/archive/id/draft-ietf-tsvwg-l4s-arch-20.html#section-5.1-1.2.5.1)

**Comment by sginoza** [comment]
another example: 
Section 1.1 of draft-ietf-rats-uccs-12 contains a block quote from [NIST-SP800-90Ar1] (see definition for Secure Channel) within a <dd>.  

https://www.ietf.org/archive/id/draft-ietf-rats-uccs-12.html#section-1.1

**Comment by kesara** [comment]
`blockquote` is only allowed in `section` as described in RFC 7991[^1].

But `xml2rfc` allows `blockquote` in `aside` and `li` as well[^2].

This needs to be addressed in RSWG correct? 

[^1]: https://www.rfc-editor.org/rfc/rfc7991.html#section-2.10
[^2]: https://www.ietf.org/archive/id/draft-levkowetz-xml2rfc-v3-implementation-notes-13.html#section-5.4

**Comment by cabo** [comment]
> This needs to be addressed in RSWG correct?

At some point, yes.

The best way to handle this probably is to generate a fixed grammar (e.g., here, allowing all block-level elements in all places where block-level elements are allowed), agree this as an RPC input after soliciting community feedback and building a prototype implementation in xml2rfc, get a nod from RSWG, and release the xml2rfc update.

**Comment by JayDaley** [comment]
> The best way to handle this probably is to generate a fixed grammar (e.g., here, allowing all block-level elements in all places where block-level elements are allowed), agree this as an RPC input after soliciting community feedback and building a prototype implementation in xml2rfc, get a nod from RSWG, and release the xml2rfc update.

I agree, though I don't think the "get a nod from RSWG" step is needed.

**Comment by kesara** [comment]
See the draft PR #1239.

Example with `blockquote` inside `dd`: [XML](https://t4.fq.nz/xml2rfc.1239/draft-doe-dd-blockquote-00.html), [Text](https://t4.fq.nz/xml2rfc.1239/draft-doe-dd-blockquote-00.txt), [HTML](https://t4.fq.nz/xml2rfc.1239/draft-doe-dd-blockquote-00.html), [PDF](https://t4.fq.nz/xml2rfc.1239/draft-doe-dd-blockquote-00.pdf)

**Comment by rjsparks** [comment]
#1239 only addresses blockquote inside dd, correct? Shouldn't this go all the way to 
> allowing all block-level elements in all places where block-level elements are allowed

**Comment by kesara** [comment]
I'm in favour of small changes every change comes with it's own set of complexities to get the HTML, TXT renderings correct.

**Comment by rjsparks** [comment]
Maybe we need a master issue with a tasklist for all the things that need to change that we can hang a series of PRs off of?

**Comment by cabo** [comment]
> I'm in favour of small changes every change comes with it's own set of complexities to get the HTML, TXT renderings correct.

Indeed, each time we expand the grammar we need to check that the renderers handle the expanded grammar.
So I think Robert's comment describes the end goal, and Kesara is showing a way to get there.
As long as having started on the road to getting this right doesn't somehow stop us from getting there, I applaud Kesara's approach of little steps.
We can still sit down and do a repaired v3 grammar that we want to get to over time.

**Comment by cabo** [comment]
> Maybe we need a master issue with a tasklist for all the things that need to change that we can hang a series of PRs off of?

We might need that.
I hope you don't say we need to wait with this particular fix until we have a complete list.

**Comment by rjsparks** [comment]
no - the tasklist can be partially complete at any point in time - we don't have to wait for it to be completed before merging the individual PRs.

**Comment by cabo** [comment]
> no - the tasklist can be partially complete at any point in time - we don't have to wait for it to be completed before merging the individual PRs.

Good.  There are probably several such lists started somewhere, including https://github.com/rfc-format/draft-iab-xml2rfc-v3-bis/issues/200

**Comment by cabo** [comment]
> For instance, does it make sense to allow sourcecode in the abstract?

Yes.

<blockquote>
Abstract

This document addresses the situation where

``` javascript
two = 1 + 1
```

does not leave the variable two with the value 2.
</blockquote>

**Comment by kesara** [comment]
I haven't reviewed the issues under this [`component: v3 vocabulary`](https://github.com/ietf-tools/xml2rfc/issues?q=state%3Aopen%20label%3A%22component%3A%20v3%20vocabulary%22) yet.
May be that label can be the list of issues that needs updates to XML specification?
I think we should rename that label to something better too.
